surface/src/vtk_smoothing/vtk_utils.cpp

```diff
@@ -48,6 +48,7 @@
 #include <vtkPolyData.h>
 #include <vtkPointData.h>
 #include <vtkFloatArray.h>
+#include <vtkUnsignedCharArray.h>
```

Comment:
This include looks suspicious (other places have `vtkIdTypeArray.h` instead)

Reply:
Are you suggesting that we change this:
`  vtkUnsignedCharArray* poly_colors = nullptr;
  if (poly_data->GetPointData() != nullptr)
    poly_colors = vtkUnsignedCharArray::SafeDownCast (poly_data->GetPointData ()->GetScalars ("Colors"));`

to use vtkIdTypeArray instead of vtkUnsignedCharArray?

Should this be a different PR?  At the moment, without that include, I am getting:
error: incomplete type 'vtkUnsignedCharArray' used in nested name specifier
  107 |     poly_colors = vtkUnsignedCharArray::SafeDownCast (poly_data->GetPointData ()->GetScalars ("Colors"));

on g++ 9.3.0 on Ubuntu 20.04

Reply:
This is where the Scalar data is set:
image->AllocateScalars (VTK_UNSIGNED_CHAR, 3);

It is standard rgb color data, so looks like it is correct to have a vtkUnsignedCharArray

Reply:
The newly included header(s) coincided with the `ifdef` and I thought the header file for that type had changed. If the macro and header are independent, then it was a misunderstanding and please go ahead and resolve the conversation.